addons/godot-next/nodes/behavior_collection.gd

```diff
@@ -0,0 +1,126 @@
+tool
+extends Node
+class_name BehaviorCollection
+
+##### PROPERTIES #####
+
+var _behaviors: Dictionary = { }
+var _callbacks: Dictionary = {
+	"_ready" : {},
+	"_process" : {},
+	"_physics_process" : {},
+	"_input" : {},
+	"_unhandled_input" : {},
+	"_unhandled_key_input" : {}
+}
+
+##### NOTIFICATIONS #####
+
+func _set(property: String, value) -> bool:
+	if not Engine.editor_hint: return false
+	if PropertyUtils.is_set_element(property):
+		return PropertyUtils.set_set_element(self, property, value)
+	if PropertyUtils.is_array_element(property):
+		return PropertyUtils.set_array_element(self, property, value)
+	return false
+
+func _get(property: String):
+	if not Engine.editor_hint: return null
+	if PropertyUtils.is_set_element(property):
+		return PropertyUtils.get_set_element(self, property)
+	if PropertyUtils.is_array_element(property):
+		return PropertyUtils.get_array_element(self, property)
+	return null
+
+func _get_property_list():
+	var list = []
+	list += PropertyUtils.generate_resource_set(self, "_behaviors", Behavior)
+	return list
+
+func _ready():
```

Comment:
You seem to have forgotten to do `if behavior.enabled:` for a lot of these. If not for use with these callbacks, what other reason would one have to keep that property in the Behavior class?